src/protocol/Checkpoint.sol

```diff
@@ -4,7 +4,7 @@ pragma solidity ^0.8.28;
 import {IDataFeed} from "./IDataFeed.sol";
 import {IVerifier} from "./IVerifier.sol";
 
-contract Inbox {
+contract Checkpoint {
```

Comment:
I would reconsider renaming everything inside that we're calling a `checkpoint`. Given the contract itself is called `checkpoint`, everything else may be deemed redundant